Test domain tags names, factor InverseJacobian into simple and compute tags

nilsdeppe opened the pull request:
## Proposed changes

- Test domain tags name functions
- Include target frame in element map tag
- Have the `InverseJacobian` tag be a simple tag and add a compute tag.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Make InverseJacobian tag a simple tag and add compute tag

nilsdeppe committed:
Test domain tag names

nilsdeppe committed:
Add frame to ElementMap tag name

kidder commented on tests/Unit/Domain/Test_Tags.cpp:
```diff
@@ -0,0 +1,36 @@
+namespace {
+template <size_t Dim>
+void test() noexcept {
+  CHECK(db::tag_name<Tags::Domain<Dim>>() == "Domain");
```
for simple tags please use `TestHelpers::db::test_simple_tag<Tag>("ExpectedNameOfTag");` by including `tests/Unit/DataStructures/DataBox/TestHelpers.hpp`

kidder replied:
please remove the corresponding check in `Test_Domain.cpp`

kidder commented on tests/Unit/Domain/Test_Tags.cpp:
```diff
@@ -0,0 +1,36 @@
+  CHECK(db::tag_name<Tags::InitialExtents<Dim>>() == "InitialExtents");
+  CHECK(db::tag_name<Tags::InitialRefinementLevels<Dim>>() ==
+        "InitialRefinementLevels");
+  CHECK(db::tag_name<Tags::Element<Dim>>() == "Element");
```
please remove the corresponding check in `Test_Element.cpp`

kidder commented on tests/Unit/Domain/Test_Tags.cpp:
```diff
@@ -0,0 +1,36 @@
+        "InitialRefinementLevels");
+  CHECK(db::tag_name<Tags::Element<Dim>>() == "Element");
+  CHECK(db::tag_name<Tags::Mesh<Dim>>() == "Mesh");
+  CHECK(db::tag_name<Tags::ElementMap<Dim>>() == "ElementMap(Inertial)");
```
ditto for `Test_ElementMap.cpp`

kidder commented on tests/Unit/Domain/Test_Tags.cpp:
```diff
@@ -0,0 +1,36 @@
+        "ElementMap(Grid)");
+  CHECK(db::tag_name<Tags::Coordinates<Dim, Frame::Grid>>() ==
+        "GridCoordinates");
+  CHECK(db::tag_name<Tags::Coordinates<Dim, Frame::Logical>>() ==
```
ditto for Test_CoordinatesTag.cpp`

kidder replied:
ping

kidder commented on src/Domain/Tags.hpp:
```diff
@@ -161,15 +161,33 @@ struct MappedCoordinates
+      db::ComputeTag,
+      db::PrefixTag {
   using tag = MapTag;
-  static std::string name() noexcept { return "InverseJacobian"; }
```
you need a `base` type alias or a `name` function for `db::tag_name` to work for a compute item

kidder commented on src/Domain/Tags.hpp:
```diff
@@ -161,15 +161,33 @@ struct MappedCoordinates
+                      typename db::const_item_type<MapTag>::source_frame,
+                      typename db::const_item_type<MapTag>::target_frame>,
+      db::ComputeTag,
+      db::PrefixTag {
```
why is the compute tag a `db::PrefixTag` but not the simple tag?

nilsdeppe replied:
Presumably because the compute tag is applied to `MapTag`, but I don't think that agrees with the way we originally wanted prefix tags to work. I think we can safely remove `PrefixTag` in this case. What do you think?

kidder replied:
If it can be removed, I would just remove it.

nilsdeppe replied:
Okay, I added a new commit that removes it

kidder commented on src/Domain/Tags.hpp:
```diff
@@ -161,15 +161,33 @@ struct MappedCoordinates
 
+/// \ingroup DataBoxTagsGroup
+/// \ingroup ComputationalDomainGroup
+/// Computes the inverse Jacobian from the source frame to the target frame.
```
the simple tag doesn't compute anything

nilsdeppe committed:
Remove PrefixTag from InverseJacobianCompute

nilsdeppe commented:
Fixups pushed, some things basically had to get squashed immediately because of conflicts between commits.

moxcodes commented on tests/Unit/Domain/Test_Tags.cpp:
```diff
@@ -31,9 +31,9 @@ void test() noexcept {
   TestHelpers::db::test_simple_tag<
       Tags::InverseJacobian<Dim, Frame::Logical, Frame::Inertial>>(
       "InverseJacobian(Logical,Inertial)");
-  TestHelpers::db::test_simple_tag<Tags::InverseJacobianCompute<
```
why is this change reversed?

nilsdeppe replied:
Because the function `test_simple_tag` doesn't work on compute tags and the `test_compute_tag` function isn't in develop yet. It's in Larry's tag PR :)

moxcodes reviewed (commented):
Apart from the single comment on the fixup, I'm happy for you to squash once @kidder is happy.

kidder commented:
you can squash (including the ping)

nilsdeppe commented:
Rebased and squashed. I had to update `Test_SemidiscretizedDg.cpp` for the wave equation. This is in the commit `Make InverseJacobian tag a simple tag and add compute tag`
